skbio/tree/_nj.py

```diff
@@ -109,11 +118,15 @@ def nj(dm, disallow_negative_branch_length=True, result_constructor=None):
             "Distance matrix must be at least 3x3 to "
             "generate a neighbor joining tree."
         )
-
+    # default result constructor where root is placed at midpoint of farthest OTUs
     if result_constructor is None:
 
         def result_constructor(x):
-            return TreeNode.read(io.StringIO(x), format="newick")
+            y = TreeNode.read(io.StringIO(x), format="newick")
```

Comment:
Can this change in behavior be parameterized defaulting to the existing behavior? Otherwise, the output from this method is not identical to prior versions of scikit-bio which would trigger a minor version change

Reply:
Agreed. I would suggest making midpoint rooting an after step. To my knowledge, other programs also don't do midpoint rooting by default.